### PR TITLE
Video streaming improvments

### PR DESCRIPTION
**Pull request by AlexeyBulatov**
This PR continues work on messages for video streaming.
It adds following:
1. 2 new parameters for video: ```bit rate``` and ```video image rotation```
2. new parameter ```video format ID```
3. messages for setting parameters of video streaming target address ```VIDEO_STREAM_TARGET```
4. command for request this options ```MAV_CMD_REQUEST_VIDEO_STREAM_TARGET```
5. fix some typos

@LorenzMeier @julianoes please review

### CODE REVIEW

**Comment by magicrub** [comment]
Please remove the WIP text.

**Comment by magicrub** [comment]
Ahhh I just noticed there was existing WIP text. Nevermind.

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -1382,13 +1382,21 @@
         <param index="2">Frames per second</param>
         <param index="3">Resolution in megapixels (0.3 for 640x480, 1.3 for 1280x720, etc), set to 0 if param 4/5 are used</param>
         <param index="4">WIP: Resolution horizontal in pixels</param>
-        <param index="5">WIP: Resolution horizontal in pixels</param>
+        <param index="5">WIP: Resolution vertical in pixels</param>
+        <param index="6">WIP: Bit rate (0 for auto)</param>
```
Comment (julianoes) [inline message_definitions/v1.0/common.xml]: Can you specifiy the unit please, e.g. bits/second?

**Comment by AlexeyBulatov** [comment]
Fixed
Thanks for review @julianoes

**Comment by julianoes** [comment]
@LorenzMeier this looks good to me.

**Comment by dogmaphobic** [comment]
I have a few comments:

`MAV_CMD_VIDEO_START_CAPTURE`
`MAV_CMD_VIDEO_STOP_CAPTURE`

These would be better renamed to:

`MAV_CMD_VIDEO_START_RECORDING`
`MAV_CMD_VIDEO_STOP_RECORDING`

This would disambiguate their meaning. In addition, I would add two new messages:

`MAV_CMD_VIDEO_START_STREAMING`
`MAV_CMD_VIDEO_STOP_STREAMING`

This makes it clear we are talking about two different things. Note that this is only relevant for protocols where the vehicle initiates the stream. For protocols such as RTSP, these messages are not needed as the client (GCS) initiates the stream.

While at it, I would add a couple of fields to `CAMERA_CAPTURE_STATUS`.

    uint32_t  recording_time  Time in seconds since recording started
    float  available_storage  Available (free) storage in MiB

There should also be an option to receive `CAMERA_CAPTURE_STATUS` automatically and continuously. There should also be a way to define the frequency of these messages. This is when sending `MAV_CMD_VIDEO_START_RECORDING` and `MAV_CMD_IMAGE_START_CAPTURE` (when "Number of images to capture” is greater than 1). The messages would stop coming when video recording is stopped.

For video streaming:

`SET_VIDEO_STREAM_TARGET`
`VIDEO_STREAM_TARGET`

Instead of an IP address and port number, it should be an URL defining the protocol, address and port all in one line. Port number is optional as some protocols have default port numbers.

This needs a bit more thinking as some protocols are in one direction while others are in the opposite direction. For example, RTSP, the client requests the vehicle an URL for it to connect to (client calls vehicle). For RPT/UDP, the vehicle initiates the stream targeting the client (vehicle “calls” client).

Unless I missed it, I don’t see a way to tell the format of the video stream. At minimum the resolution and frame rate. Encoding is a bit trickier as it is protocol dependent. RTSP encapsulates the protocol within its payload while RPT/UDP, you must define the encoding yourself, though is it fair to assume h.264 as the default.

Notes:

Regarding capture devices (for recording), mode switch takes time. That is, when switching from Video to Still for example, the camera may take upwards of 2 seconds to complete the switch.

Most cameras cannot shoot video when in Still mode.

Some cameras allow shooting stills when in Video mode but more often than not, the image will be captured using the current video resolution, regardless how you setup the resolution for stills.

Most cameras and/or camera controllers have their own clock. For those that don’t, there should be a message to set the current time so the image time stamps are correct.

If you want geolocation embedded in still frames, there should be a way to feed the camera and/or controller the camera’s current coordinates.

With all that said, in addition to these MAVLink messages, there should be a standard way to query camera capabilities. This information is too much to fit into MAVLink messages. The GCS needs to know ahead of time what the camera is capable of. For example, a list of all resolutions and frame rates it can record video. A list of possible ISO settings, A list of shutter speed settings, etc. The GCS needs this information in order to present the user a proper user interface and parameter validation.

These are the normal parameters needed for setting up a camera (settings to be presented to the user within a camera setup UI):

- Resolution (a list of possible values)
- Frame Rate (a list of possible values)
- Still format (Raw, Jpeg, Raw+Jpeg)
- Automatic Exposure (AE On/Off)
- Metering (Center, Spot, Average, etc.)
- Exposure compensation (a list of possible values such as “-2.0, -1.5, -1.0, -0.5, 0, 0.5, 1.0, 1.5, 2.0”)
- Shutter Speed (a list of possible values)
- ISO (a list of possible values)
- White Balance (a list of possible values)
- Audio recording on/off

**Comment by dogmaphobic** [comment]
One more thing, most cameras used in this context have a different Angle Of View (AOV) for Video and Stills. When providing the camera's lens focal length and/or AOV, there should be a differentiation.

I also realized I completely highjacked this PR with way more than it was originally intended :)

**Comment by dogmaphobic** [comment]
I added a PR with some of the above (only related to Video/Photo capture)

https://github.com/mavlink/mavlink/pull/675

**Pull request closed without merging**